tests/unit/utilities/walk-up-path-test.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+var expect = require('chai').expect;
+var walkUp = require('../../../lib/utilities/walk-up-path');
+
+describe('walk-up-path', function() {
+  it('walks up paths', function() {
+    var paths = walkUp('app/templates/my/test.hbs');
+
+    expect(paths).to.deep.equal([
+      'app/templates/my',
+      'app/templates',
+      'app',
+    ]);
+  });
+
+  it('walks up paths beginning with a slash', function() {
```

Comment:
Add a test for `/foo/foo/file.txt` to protect us from the bug I accidentally included in my first bit of feedback.